pkg/subctl/cmd/join.go

```diff
@@ -221,6 +226,30 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	exitOnError("Error deploying Submariner", err)
 }
 
+func checkRequirements(config *rest.Config) error {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return errors.WithMessage(err, "error creating API server client")
+	}
+	serverVersion, err := clientset.Discovery().ServerVersion()
+	if err != nil {
+		return errors.WithMessage(err, "error obtaining API server version")
+	}
+	major, err := strconv.Atoi(serverVersion.Major)
+	if err != nil {
+		return errors.WithMessagef(err, "error parsing API server major version %v", serverVersion.Major)
+	}
+	minor, err := strconv.Atoi(serverVersion.Minor)
+	if err != nil {
+		return errors.WithMessagef(err, "error parsing API server minor version %v", serverVersion.Minor)
+	}
+	if major < minK8sMajor || (major == minK8sMajor && minor < minK8sMinor) {
+		exitWithErrorMsg(fmt.Sprintf("Submariner requires Kubernetes %d.%d; your cluster is running %s.%s",
+			minK8sMajor, minK8sMinor, serverVersion.Major, serverVersion.Minor))
```

Comment:
Not trying to nitpick but I think we should be consistent with the other code paths (ie L232, L236 etc) and either return the error here or `exitWithErrorMsg` in the other code paths and don't return error. The latter is simpler.

Reply:
Right, I misunderstood your initial comment about `exitWithErrorMsg` ;-).

Reply:
I think the mistake here is trying to use errors for both technical errors and requirement errors. The requirements check should return something else, and use errors for only technical errors.

Reply:
yeah that makes sense - use regular output for requirement errors.